holmes/plugins/toolsets/opensearch/opensearch_traces.py

```diff
@@ -0,0 +1,198 @@
+            )
+            logging.debug(f"opensearch traces search query: {full_query}")
+            headers = {"Content-Type": "application/json"}
+            headers.update(add_auth_header(self.toolset.config.opensearch_auth_header))
+
+            logs_response = requests.get(
+                url=self.toolset.config.opensearch_traces_url,
+                timeout=180,
+                verify=True,
+                data=json.dumps(full_query),
+                headers=headers,
+            )
+            if logs_response.status_code > 300:
+                err_msg = logs_response.text
+
+            logs_response.raise_for_status()
+            return json.dumps(logs_response.json())
+        except requests.Timeout:
+            logging.warn(
+                "Timeout while fetching opensearch traces search", exc_info=True
+            )
+            return (
+                f"Request timed out while fetching opensearch traces search {err_msg}"
+            )
+        except RequestException as e:
+            logging.warn("Failed to fetch opensearch traces search", exc_info=True)
+            return f"Network error while opensearch traces search {err_msg} : {str(e)}"
+        except Exception as e:
+            logging.warn("Failed to process opensearch traces search ", exc_info=True)
+            return f"Unexpected error {err_msg}: {str(e)}"
+
+    def get_parameterized_one_liner(self, params) -> str:
+        return f'search traces: query="{params.get("query")}"'
+
+
+class OpenSearchTracesToolset(Toolset):
+    def __init__(self):
+        super().__init__(
+            name="opensearch/traces",
+            description="OpenSearch integration to fetch traces",
+            docs_url="",
+            icon_url="",
+            prerequisites=[CallablePrerequisite(callable=self.prerequisites_callable)],
+            tools=[
+                GetTracesFields(toolset=self),
+                TracesSearchQuery(toolset=self),
+            ],
+            tags=[
+                ToolsetTag.CORE,
+            ],
+        )
+        self._load_llm_instructions(
+            os.path.abspath(
+                os.path.join(
+                    os.path.dirname(__file__), "opensearch_traces_instructions.jinja2"
+                )
+            )
+        )
+
+    def prerequisites_callable(self, config: dict[str, Any]) -> bool:
+        if not config and not os.environ.get("OPENSEARCH_TRACES_URL", None):
+            return False
+        elif not config and os.environ.get("OPENSEARCH_TRACES_URL", None):
+            self.config = OpenSearchTracesConfig(
+                opensearch_traces_url=os.environ.get("OPENSEARCH_TRACES_URL"),
+                opensearch_auth_header=os.environ.get(
+                    "OPENSEARCH_LOGS_AUTH_HEADER", None
+                ),
+            )
+            return True
+        else:
+            self.config = OpenSearchTracesConfig(**config)
+            return True
```

Comment:
same comments as the logs toolset re. docs and icon urls, `get_example_config` and live health check to test the connection.

Reply:
fixed